### PR TITLE
refactor (material/core): create new per-component focus indicator customization styles in the component-level stylesheets for the old and MDC library.

### PR DESCRIPTION
**Pull request by jackierchu**
Per step 2 of HCM Focus Indicator DD (http://shortn/_QfjugJram1), this PR will:

1. Create and add new per-component customizations for all appropriate stylesheets within every component
2. Utilized the newly created focus indicator width variables (mat-focus-indicator-border-width and default-focus-indicator-width) for the per-component customizations (such as margin) as configured by the mixin. 
3. This PR no longer uses the $border-width variable and config object. Instead, this PR uses the CSS variables introduced in above for customizations. Therefore, the per-component customizations no longer depend on $border-width. Rather, it depends on its equivalent CSS variable for $border-width that was created in step 1 of the HCM Focus Indicator DD.

### CODE REVIEW

File: src/material-experimental/mdc-button/button.scss [hunk 1]
```diff
@@ -106,3 +106,26 @@
   right: $offset;
   border-width: $offset;
 }
+
+// By default, all focus indicators are flush with the bounding box of their
+// host element. For the button element, default inset/offset
```
Comment (zelliott) [inline src/material-experimental/mdc-button/button.scss:111]: I don't think we need to repeat the "By default, all focus indicators are flush with the bounding box of their host element." blurb across all components, I'd vote to remove it.

File: src/material-experimental/mdc-button/button.scss [hunk 2]
```diff
@@ -106,3 +106,26 @@
...
+.mdc-fab .mat-mdc-focus-indicator::before {
+  margin: calc(-(var(--mat-mdc-focus-indicator-border-width, $default-focus-indicator-width) + 2px));
+}
+
+.mat-mdc-outlined-button .mat-mdc-focus-indicator::before {
+  margin: calc(-(var(--mat-mdc-focus-indicator-border-width, $default-focus-indicator-width) + 3px));
+}
+
+// Render the focus indicator on focus. Defining a pseudo element's
+// content will cause it to render.
```
Comment (zelliott) [inline src/material-experimental/mdc-button/button.scss:125]: I don't think we need to repeat this comment across all components either.

File: src/material/button/button.scss [hunk 3]
```diff
@@ -153,6 +153,22 @@
...
+// contrastive and renders appropriately.
+.mat-focus-indicator.mat-flat-button::before,
+.mat-focus-indicator.mat-raised-button::before,
+.mat-focus-indicator.mat-fab::before,
+.mat-focus-indicator.mat-mini-fab::before {
+  margin: calc(-(var(--mat-focus-indicator-border-width, $default-focus-indicator-width) + 2px));
+}
+
+.mat-focus-indicator.mat-stroked-button::before
+{
```
Comment (zelliott) [inline src/material/button/button.scss:168]: This formatting doesn't look quite right, I'd expect this `{` to be on the same line above.

**Review by zelliott — COMMENTED** [review]
I think this looks really good! The diff is a little off - it looks like it includes changes from https://github.com/angular/components/pull/24524, it'd be great to get that fixed.

Also - at this point I think we should be able to run the Angular dev demo app and see focus indicators working (maybe)? There should still be double focus indicators in HCM, but it might be a good idea to run the dev demo app and see what the actual state of things are.

File: src/material-experimental/mdc-chips/chip.scss [hunk 4]
```diff
@@ -206,3 +206,24 @@
...
+// MDC sets a padding a on the chip button which stretches out the focus indicator.
+.mat-mdc-focus-indicator.mat-mdc-chip-remove::before {
+  left: 8px;
+  right: 8px;
+}
+
+// In the chips the individual actions have focus so we target a different element.
+.mat-mdc-chip-action:focus .mat-mdc-focus-indicator::before {
+  content: '';
+}
```
Comment (zelliott) [inline src/material-experimental/mdc-chips/chip.scss:229]: nit: Missing newline at EOF

**Review by zelliott — APPROVED** [review]
LGTM!

**Comment by jackierchu** [comment]
Took a look at the `angular:focus-indicators` branch and noticed the branch was missing the changes from my previous PR: https://github.com/angular/components/pull/24524 (which created new structural focus indicator styles, updated core stylesheet, removed theme stylesheets, etc). Although that PR was merged into the `angular:focus-indicators` branch, a force push occurred here in this PR (possibly without fetching from the angular/focus-indicators branch and its changes beforehand) which overrode the changes from my previous PR with the master branch. In turn, this is causing the merge conflicts here in this PR. Is someone with write-access able to recover changes from the previous PR (https://github.com/angular/components/pull/24524) and revert the changes back into the `angular:focus-indicators` branch? Thanks so much for the help and time!

**Review by crisbeto — APPROVED** [review]
LGTM

**Pull request merged**